src/patients/patient-slice.ts

```diff
@@ -318,6 +351,80 @@ export const addDiagnosis = (
   }
 }
 
+function validatePhoneNumber(phoneNumber: PhoneNumber) {
+  const error: AddPhoneNumberError = {}
+
+  if (!phoneNumber.phoneNumber) {
+    error.name = 'patient.phoneNumber.error.phoneNumberRequired'
+  }
+
+  return error
+}
+
+export const addPhoneNumber = (
+  patientId: string,
+  phoneNumber: PhoneNumber,
+
+  onSuccess?: (patient: Patient) => void,
+): AppThunk => async (dispatch) => {
+  const newPhoneNumberError = validatePhoneNumber(phoneNumber)
+  if (_.isEmpty(newPhoneNumberError)) {
+    const patient = await PatientRepository.find(patientId)
+    const phoneNumbers = patient.phoneNumber || []
+    phoneNumbers.push({ id: uuid(), ...phoneNumber })
+    patient.phoneNumber = phoneNumbers
+
+    await dispatch(updatePatient(patient, onSuccess))
+  } else {
+    newPhoneNumberError.message = 'patient.phoneNumbers.error.unableToAdd'
+    dispatch(addPhoneNumberError(newPhoneNumberError))
```

Comment:
It looks like there should be some errors being displayed with this error being dispatched, however, I don't see that functionality when using the UI.